framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2363,7 +2363,13 @@ protected void checkTypecastSafety(TypeCastTree typeCastTree) {
       return;
     }
     AnnotatedTypeMirror castType = atypeFactory.getAnnotatedType(typeCastTree);
+    if (!atypeFactory.isRelevant(castType)) {
```

Comment:
How did you decide which checks to skip if a type isn't relevant? Would is be better to change isSubtype instead?

Reply:
Most calls to `isSubtype()` are to the version in `QualifierHierarchy`, which takes `AnnotationMirror`s as arguments.  That `isSubtype()` cannot behave differently based on the Java basetype.  So, the relevance test must be done before `QualifierHierarchy.isSubtype()` is called.  I put tests in places that seemed right, based on examining all occurrences of `qualifierHierarchy.isSubtype()`.

To improve uniformity and reduce the chance of errors, we could do one of these:
 * Pass the `AnnotatedTypeMirror`s into `QualifierHierarchy.isSubtype()` as an extra argument.
 * Discourage/prevent clients from calling `QualifierHierarchy.isSubtype()`, as in `myTypeFactory.getQualifierHierarchy().isSubtype()`.  Instead, have a method `AnnotatedTypeFactory.isSubtype()` that takes `AnnotatedTypeMirror`s, checks relevance, and then delegates to the `QualifierHierarchy`.
 * Keep it as is, and prevent/fix problems through code review and testing.
 * Other ideas?

All that said, I did change the `isSubtype()` that takes annotated types as well.